### PR TITLE
Device Metadata API

### PR DESCRIPTION
**Pull request by russellb**
- docs: Propose API support for device metadata
- api: WIP support for DeviceMetadata API

Issue #1171

### CODE REVIEW

**Comment by chirino** [comment]
rebased off main

File: cmd/nexctl/device_metadata.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	return fmt.Errorf("not implemented")
+}
+
+func deleteDeviceMetadata(c *cli.Context, deviceID uuid.UUID, key string) error {
+	return fmt.Errorf("not implemented")
+}
+
+func clearDeviceMetadata(c *cli.Context, deviceID uuid.UUID) error {
+	return fmt.Errorf("not implemented")
+}
```
Comment (russellb) [inline cmd/nexctl/device_metadata.go:169]: TODO -- finish nexctl commands

File: docs/development/design/device-metadata.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Instead of changing the Device table, we will create a new table called DeviceMetadata. I propose keeping it separate from the Device table while we work through how we want this to work. This table will have the following columns:
+
+```go
+type Base struct {
+    ID        uuid.UUID      `gorm:"type:uuid;primary_key;" json:"id" example:"aa22666c-0f57-45cb-a449-16efecc04f2e"`
+    CreatedAt time.Time      `json:"-"`
+    UpdatedAt time.Time      `json:"-"`
+    DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
+}
```
Comment (russellb) [inline docs/development/design/device-metadata.md:19]: TODO -- update this design doc to reflect where we ended up

**Review by russellb — COMMENTED** [review]
just noting a couple todo items

File: cmd/nexctl/device_metadata.go (hunk 1 above)
Comment (chirino) [reply cmd/nexctl/device_metadata.go]: done now.

**Comment by chirino** [comment]
rebased off main

**Review by nerdalert — APPROVED** [review]
LGTM

**Pull request merged**